pkg/controller/core/dynamicresourceallocationconfig_controller.go

```diff
@@ -0,0 +1,90 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/features"
+)
+
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=dynamicresourceallocationconfigs,verbs=get;list;watch
+
+// DynamicResourceAllocationConfigReconciler keeps the cache's DeviceClass→logical-resource
+// mapping in sync with the singleton DynamicResourceAllocationConfig object.
+// It is only registered when the DynamicResourceAllocation feature gate is enabled.
+type DynamicResourceAllocationConfigReconciler struct {
+	client    client.Client
+	cache     *cache.Cache
+	log       logr.Logger
+	namespace string
+}
+
+func NewDRAConfigReconciler(client client.Client, c *cache.Cache, namespace string) *DynamicResourceAllocationConfigReconciler {
+	return &DynamicResourceAllocationConfigReconciler{
+		client:    client,
+		cache:     c,
+		log:       ctrl.Log.WithName("dra-config-reconciler"),
+		namespace: namespace,
+	}
+}
+
+func (r *DynamicResourceAllocationConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return ctrl.Result{}, nil
+	}
+
+	var cfg kueuealpha.DynamicResourceAllocationConfig
+	if err := r.client.Get(ctx, req.NamespacedName, &cfg); err != nil {
+		// TODO: this should never happen. But if a user deletes the singleton DRA by mistake, we should
+		//  have a way to gracefully handle it. May be adding a finalizer to the singleton DRAConfig object?
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.log.V(2).Info("Reconciling DynamicResourceAllocationConfig", "name", req.NamespacedName)
+	// we do not expect a lot of reconciles triggered here, therefore directly updating the cache should be okay
+	r.cache.AddOrUpdateDynamicResourceAllocationConfig(r.log, &cfg)
+	return ctrl.Result{}, nil
+}
+
+func (r *DynamicResourceAllocationConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("dynamicresourceallocationconfig_controller").
+		For(&kueuealpha.DynamicResourceAllocationConfig{}).
+		WithEventFilter(predicate.Funcs{
+			// Only singleton named "default" in the configured namespace matters.
+			CreateFunc: func(e event.CreateEvent) bool {
+				obj := e.Object.(*kueuealpha.DynamicResourceAllocationConfig)
+				return obj.Name == "default" && obj.Namespace == r.namespace
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				obj := e.ObjectNew.(*kueuealpha.DynamicResourceAllocationConfig)
+				return obj.Name == "default" && obj.Namespace == r.namespace
+			},
```

Comment:
No need for the checks. We can ensure this is singleton using CEL, so that inside the codebase we can just rely on that.